CharlieBackend.Business/Services/HomeworkStudentService.cs

```diff
@@ -159,33 +169,91 @@ public async Task<IList<HomeworkStudentDto>> GetHomeworkStudentForMentor(long ho
             return _mapper.Map<IList<HomeworkStudentDto>>(homeworkStudent);
         }
 
-        private async IAsyncEnumerable<string> ValidateHomeworkStudentRequest(HomeworkStudentRequestDto homeworkStudent, Student student , Homework homework)
+        public async Task<Result<HomeworkStudentDto>> UpdateMarkAsync(UpdateMarkRequestDto request)
         {
-            
-            if (homeworkStudent == default)
+            var homeworkStudent = await _unitOfWork.HomeworkStudentRepository.GetByIdAsync(request.StudentHomeworkId);
+
+            long accountId = _currentUserService.AccountId;
+
+            if (homeworkStudent.Mark == null)
             {
-                yield return "Please provide request data";
-                yield break;
+                homeworkStudent.Mark = new Mark
+                {
```

Comment:
To avoid code duplication, in the if statement just assign a new Mark().  
`if(homeworkStudent.Mark == null) 
{
    homeworkStudent.Mark = new Mark();
} 
 homeworkStudent.Mark.Value = request.StudentMark;
                homeworkStudent.Mark.Comment = request.MentorComment;
                homeworkStudent.Mark.EvaluationDate = DateTime.UtcNow;
                homeworkStudent.Mark.Type = request.MarkType;
                homeworkStudent.Mark.EvaluatedBy = accountId;
`